lib/questions.rb

```diff
@@ -155,30 +187,38 @@ def get_domain_name_from_email_address(email)
 # 'the lion the witch and the wardrobe' becomes
 # 'The Lion the Witch and the Wardrobe'
 def titleize_a_string(string)
+  string.split.map{|word| ['and','the'].include?(word) ? word : word.capitalize}.join(" ").sub(/./, &:upcase)
 end
 
 # return true if a string contains any special characters
 # where 'special character' means anything apart from the letters
 # a-z (uppercase and lower) or numbers
 def check_a_string_for_special_characters(string)
+  /\W/ === string
 end
 
 # get the upper limit of a range. e.g. for the range 1..20, you
 # should return 20
 def get_upper_limit_of(range)
+  range.max
 end
 
 # should return true for a 3 dot range like 1...20, false for a 
 # normal 2 dot range
 def is_a_3_dot_range?(range)
+  range.max != range.last
 end
 
 # get the square root of a number
 def square_root_of(number)
+  number ** 0.5
 end
 
 # count the number of words in a file
 def word_count_a_file(file_path)
+  word_count = 0
+  File.open(file_path, "r") {|file| file.each_line { |line| words = line.split(' ').each { |word| word_count += 1}}}
```

Comment:
Unused block argument - `word`. You can omit the argument if you don't care about it.<br>Line is too long. [116/80]<br>Space between { and | missing.<br>Space missing inside }.